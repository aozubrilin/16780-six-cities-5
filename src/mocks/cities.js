export const CityCoordinate = {
  BRUSSELS: {coordinates: [50.85, 4.3]},
  PARIS: {coordinates: [48.86, 2.34]},
  AMSTERDAM: {coordinates: [52.38333, 4.9]},
  COLOGNE: {coordinates: [50.933, 6.95]},
  HAMBURG: {coordinates: [53.56, 10.015]},
  DUSSELDORF: {coordinates: [51.22, 6.78]},
};
