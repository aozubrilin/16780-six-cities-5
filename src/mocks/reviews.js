export default [
  {
    id: 0,
    name: `Max`,
    avatar: `img/avatar-max.jpg`,
    reiting: 1,
    date: new Date(`2019-04-18`),
    text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Phasellus egestas tellus rutrum tellus.`,
  },
  {
    id: 1,
    name: `Angelina`,
    avatar: `img/avatar-angelina.jpg`,
    reiting: 2,
    date: new Date(`2020-06-18`),
    text: `Volutpat lacus laoreet non curabitur gravida arcu. Morbi blandit cursus risus at ultrices mi tempus. Consectetur adipiscing elit pellentesque habitant morbi. Fermentum leo vel orci porta.`,
  },
  {
    id: 2,
    name: `Ann`,
    avatar: `img/avatar.svg`,
    reiting: 3,
    date: new Date(`2020-06-17`),
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
  {
    id: 3,
    name: `Bob`,
    avatar: `img/avatar.svg`,
    reiting: 4,
    date: new Date(`2020-09-17`),
    text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  }
];
