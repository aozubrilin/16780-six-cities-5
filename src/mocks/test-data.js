export const offers = [
  {
    bedroomsCount: 3,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    features: [`Towels`, `Baby seat`, `Laptop friendly workspace`, `Dishwasher`, `Washing machine`, `Air conditioning`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 2,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: false,
    isPremium: false,
    coordinates: [50.839557, 4.346697],
    adultsCount: 8,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    price: 323,
    rating: 4.3,
    title: `Amazing and Extremely Central Flat`,
    type: `Apartment`
  },
  {
    bedroomsCount: 4,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    features: [`Washer`, `Laptop friendly workspace`, `Breakfast`, `Air conditioning`, `Baby seat`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 3,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: false,
    isPremium: false,
    coordinates: [50.865556999999995, 4.371696999999999],
    adultsCount: 7,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    price: 218,
    rating: 4.6,
    title: `The house among olive `,
    type: `Apartment`
  },
  {
    bedroomsCount: 1,
    city: {
      name: `Amsterdam`,
      coordinates: [52.37454, 4.897976],
      zoom: 13
    },
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    features: [`Breakfast`, `Air conditioning`, `Laptop friendly workspace`, `Baby seat`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 22,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`
    ],
    isFavorite: false,
    isPremium: true,
    coordinates: [52.364540000000005, 4.9019759999999994],
    adultsCount: 3,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    price: 175,
    rating: 3.9,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Room`,
  }
];

export const favoriteOffers = [
  {
    bedroomsCount: 3,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    features: [`Towels`, `Baby seat`, `Laptop friendly workspace`, `Dishwasher`, `Washing machine`, `Air conditioning`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 2,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: true,
    isPremium: false,
    coordinates: [50.839557, 4.346697],
    adultsCount: 8,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    price: 323,
    rating: 4.3,
    title: `Amazing and Extremely Central Flat`,
    type: `Apartment`
  },
  {
    bedroomsCount: 4,
    city: {
      name: `Brussels`,
      coordinates: [50.846557, 4.351697],
      zoom: 13
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    features: [`Washer`, `Laptop friendly workspace`, `Breakfast`, `Air conditioning`, `Baby seat`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 3,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    isFavorite: true,
    isPremium: false,
    coordinates: [50.865556999999995, 4.371696999999999],
    adultsCount: 7,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    price: 218,
    rating: 4.6,
    title: `The house among olive `,
    type: `Apartment`
  },
  {
    bedroomsCount: 1,
    city: {
      name: `Amsterdam`,
      coordinates: [52.37454, 4.897976],
      zoom: 13
    },
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    features: [`Breakfast`, `Air conditioning`, `Laptop friendly workspace`, `Baby seat`, `Washer`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isProUser: true,
      name: `Angelina`
    },
    id: 22,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`
    ],
    isFavorite: true,
    isPremium: true,
    coordinates: [52.364540000000005, 4.9019759999999994],
    adultsCount: 3,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    price: 175,
    rating: 3.9,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Room`,
  }
];

export const reviews = [
  {
    id: 1,
    date: `2020-11-05T13:38:44.753Z`,
    text: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    rating: 4,
    user: {
      id: 16,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`,
      isProUser: true,
      name: `Mollie`
    }
  },
  {
    id: 2,
    date: `2020-11-05T13:38:44.753Z`,
    text: `I stayed here for one night and it was an unpleasant experience.`,
    rating: 4,
    user: {
      id: 19,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
      isProUser: false,
      name: `Christina`
    }
  },
  {
    id: 3,
    date: `2020-11-05T13:38:44.753Z`,
    text: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
    rating: 5,
    user: {
      id: 12,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
      isProUser: true,
      name: `Isaac`
    }
  }
];

export const mockStore = {
  app: {
    city: `Amsterdam`,
    cities: [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`],
    currentSortType: `Popular`,
    activeOfferId: -1,
  },
  data: {
    offers: [],
    isLoadedOffers: false,
    nearOffers: [],
    isLoadedNearOffers: false,
    favoriteOffers: [],
    isLoadedfavoriteOffers: false,
    currentOffer: {},
    isLoadedCurrentOffer: false,
    errorFetchMessadge: ``,
    reviews: [],
    isLoadedRviews: false,
  },
  user: {
    authorizationStatus: `NO_AUTH`,
    userInfo: {},
  }
};
