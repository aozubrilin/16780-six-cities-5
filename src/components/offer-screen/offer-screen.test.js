import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from "react-router-dom";
import history from "../../browser-history";
import OfferScreen from "./offer-screen";
import thunk from "redux-thunk";
import {offers, reviews} from "../../mocks/test-data";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock(`../../store/api-actions`,
    () => ({
      fetchIdOffer: jest.fn(() => jest.fn(() => Promise.resolve())),
      fetchNearOffers: jest.fn(() => jest.fn(() => Promise.resolve())),
      fetchReviews: jest.fn(() => jest.fn(() => Promise.resolve())),
    })
);


describe(`OfferScreen snapshot`, () => {

  it(`OfferScreen render correctly`, () => {

    const store = mockStore({
      app: {
        activeOfferId: -1,
      },
      data: {
        nearOffers: offers,
        isLoadedNearOffers: false,
        currentOffer: offers[0],
        isLoadedCurrentOffer: false,
        errorFetchMessadge: ``,
        reviews,
        isLoadedRviews: false,
      },
      user: {
        authorizationStatus: `AUTH`,
        userInfo: {
          email: `oliver@mail.ru`,
          avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
        }
      },
    });


    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <OfferScreen
              match={{params: {id: 2}}}
            />
          </Router>
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        }
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


