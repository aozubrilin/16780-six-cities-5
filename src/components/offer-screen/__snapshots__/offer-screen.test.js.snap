// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferScreen snapshot OfferScreen render correctly 1`] = `
Array [
  "",
  <div
    className="page"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
              href="/"
              onClick={[Function]}
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="/favorites"
                  onClick={[Function]}
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__user-name user__name"
                  >
                    oliver@mail.ru
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--property"
    >
      <section
        className="property"
      >
        <div
          className="property__gallery-container container"
        >
          <div
            className="property__gallery"
          >
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="property__container container"
        >
          <div
            className="property__wrapper"
          >
            
            <div
              className="property__name-wrapper"
            >
              <h1
                className="property__name"
              >
                Amazing and Extremely Central Flat
              </h1>
              <button
                className="property__bookmark-button button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To
                   bookmarks
                </span>
              </button>
            </div>
            <div
              className="property__rating rating"
            >
              <div
                className="property__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                className="property__rating-value rating__value"
              >
                4.3
              </span>
            </div>
            <ul
              className="property__features"
            >
              <li
                className="property__feature property__feature--entire"
              >
                Apartment
              </li>
              <li
                className="property__feature property__feature--bedrooms"
              >
                3
                 Bedrooms
              </li>
              <li
                className="property__feature property__feature--adults"
              >
                Max 
                8
                 adults
              </li>
            </ul>
            <div
              className="property__price"
            >
              <b
                className="property__price-value"
              >
                €
                323
              </b>
              <span
                className="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              className="property__inside"
            >
              <h2
                className="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                className="property__inside-list"
              >
                <li
                  className="property__inside-item"
                >
                  Towels
                </li>
                <li
                  className="property__inside-item"
                >
                  Baby seat
                </li>
                <li
                  className="property__inside-item"
                >
                  Laptop friendly workspace
                </li>
                <li
                  className="property__inside-item"
                >
                  Dishwasher
                </li>
                <li
                  className="property__inside-item"
                >
                  Washing machine
                </li>
                <li
                  className="property__inside-item"
                >
                  Air conditioning
                </li>
                <li
                  className="property__inside-item"
                >
                  Washer
                </li>
              </ul>
            </div>
            <div
              className="property__host"
            >
              <h2
                className="property__host-title"
              >
                Meet the host
              </h2>
              <div
                className="property__host-user user"
              >
                <div
                  className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                >
                  <img
                    alt="Host avatar"
                    className="property__avatar user__avatar"
                    height="74"
                    src="img/avatar-angelina.jpg"
                    width="74"
                  />
                </div>
                <span
                  className="property__user-name"
                >
                  Angelina
                </span>
              </div>
              <div
                className="property__description"
              >
                <p
                  className="property__text"
                >
                  This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.
                </p>
              </div>
            </div>
            <section
              className="property__reviews reviews"
            >
              <h2
                className="reviews__title"
              >
                Reviews ·
                <span
                  className="reviews__amount"
                >
                  3
                </span>
              </h2>
              <ul
                className="reviews__list"
              >
                <li
                  className="reviews__item"
                >
                  <div
                    className="reviews__user user"
                  >
                    <div
                      className="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        className="reviews__avatar user__avatar"
                        height="54"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg"
                        width="54"
                      />
                    </div>
                    <span
                      className="reviews__user-name"
                    >
                      Mollie
                    </span>
                  </div>
                  <div
                    className="reviews__info"
                  >
                    <div
                      className="reviews__rating rating"
                    >
                      <div
                        className="reviews__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      className="reviews__text"
                    >
                      The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.
                    </p>
                    <time
                      className="reviews__time"
                      dateTime="2020-11-05T13:38:44.753Z"
                    >
                      November 2020
                    </time>
                  </div>
                </li>
                <li
                  className="reviews__item"
                >
                  <div
                    className="reviews__user user"
                  >
                    <div
                      className="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        className="reviews__avatar user__avatar"
                        height="54"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg"
                        width="54"
                      />
                    </div>
                    <span
                      className="reviews__user-name"
                    >
                      Christina
                    </span>
                  </div>
                  <div
                    className="reviews__info"
                  >
                    <div
                      className="reviews__rating rating"
                    >
                      <div
                        className="reviews__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      className="reviews__text"
                    >
                      I stayed here for one night and it was an unpleasant experience.
                    </p>
                    <time
                      className="reviews__time"
                      dateTime="2020-11-05T13:38:44.753Z"
                    >
                      November 2020
                    </time>
                  </div>
                </li>
                <li
                  className="reviews__item"
                >
                  <div
                    className="reviews__user user"
                  >
                    <div
                      className="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        className="reviews__avatar user__avatar"
                        height="54"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
                        width="54"
                      />
                    </div>
                    <span
                      className="reviews__user-name"
                    >
                      Isaac
                    </span>
                  </div>
                  <div
                    className="reviews__info"
                  >
                    <div
                      className="reviews__rating rating"
                    >
                      <div
                        className="reviews__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "100%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      className="reviews__text"
                    >
                      The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.
                    </p>
                    <time
                      className="reviews__time"
                      dateTime="2020-11-05T13:38:44.753Z"
                    >
                      November 2020
                    </time>
                  </div>
                </li>
              </ul>
              <form
                action="#"
                className="reviews__form form"
                method="post"
                onSubmit={[Function]}
              >
                <label
                  className="reviews__label form__label"
                  htmlFor="review"
                >
                  Your review
                </label>
                <div
                  className="reviews__rating-form form__rating"
                >
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="5-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value="5"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="5-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="4-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value="4"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="4-stars"
                    title="good"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="3-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value="3"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="3-stars"
                    title="not bad"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="2-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value="2"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="2-stars"
                    title="badly"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="1-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value="1"
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="1-stars"
                    title="terribly"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                </div>
                <textarea
                  className="reviews__textarea form__textarea"
                  id="review"
                  name="review"
                  onChange={[Function]}
                  placeholder="Tell how was your stay, what you like and what can be improved"
                  value=""
                />
                <div
                  className="reviews__button-wrapper"
                >
                  <p
                    className="reviews__help"
                  >
                    To submit review please make sure to set 
                    <span
                      className="reviews__star"
                    >
                      rating
                    </span>
                     and describe your stay with at least
                    <b
                      className="reviews__text-amount"
                    >
                      50 characters
                    </b>
                    .
                  </p>
                  <button
                    className="reviews__submit form__submit button"
                    disabled="disabled"
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </section>
          </div>
        </div>
        <section
          className="property__map map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </section>
      <div
        className="container"
      >
        <section
          className="near-places places"
        >
          <h2
            className="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            className="near-places__list places__list"
          >
            <article
              className="near-places__card place-card "
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              
              <div
                className="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/offer/2"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      323
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To
                       bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/2"
                    onClick={[Function]}
                  >
                    Amazing and Extremely Central Flat
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
            <article
              className="near-places__card place-card "
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              
              <div
                className="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/offer/3"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      218
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To
                       bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/3"
                    onClick={[Function]}
                  >
                    The house among olive 
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
            <article
              className="near-places__card place-card "
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/offer/22"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      175
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To
                       bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/22"
                    onClick={[Function]}
                  >
                    Beautiful & luxurious apartment at great location
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Room
                </p>
              </div>
            </article>
          </div>
        </section>
      </div>
    </main>
  </div>,
]
`;
